Docs/development/style-guide.md

```diff
@@ -0,0 +1,286 @@
+namespace StyleSheetExample
+{
+
+    // ENUMS:
+    // - Use a singular type name.
+    // - No prefix or suffix.
+    public enum Direction
+    {
+        North,
+        South,
+        East,
+        West,
+    }
+
+    // FLAGS ENUMS:
+    // - Use a plural type name 
+    // - No prefix or suffix.
+    // - Use column-alignment for binary values
+    [Flags]
+    public enum AttackModes
+    {
+        // Decimal                         // Binary
+        None = 0,                          // 000000
+        Melee = 1,                         // 000001
+        Ranged = 2,                        // 000010
+        Special = 4,                       // 000100
+
+        MeleeAndSpecial = Melee | Special  // 000101
+    }
+
+    // INTERFACES:
+    // - Name interfaces with adjective phrases.
+    // - Use the 'I' prefix.
+    public interface IDamageable
+    {
+        string DamageTypeName { get; }
+        float DamageValue { get; }
+
+        // METHODS:
+        // - Start a methods name with a verbs or verb phrases to show an action.
+        // - Parameter names are camelCase.
+        bool ApplyDamage(string description, float damage, int numberOfHits);
+    }
+
+    public interface IDamageable<T>
+    {
+        void Damage(T damageTaken);
+    }
+
+    // CLASSES or STRUCTS:
+    // - Name them with nouns or noun phrases.
+    // - Avoid prefixes.
+    // - One Class/Interface per file. The source file name must match. 
+    public class StyleExample : MonoBehaviour
+    {
+
+        // FIELDS: 
+        // - Avoid special characters (backslashes, symbols, Unicode characters); these can interfere with command line tools.
+        // - Use nouns for names, but prefix booleans with a verb.
+        // - Use meaningful names. Make names searchable and pronounceable. Don’t abbreviate (unless it’s math).
+        // - Use Pascal case for public fields. Use camel case for private variables.
+        // - Add an underscore (_) in front of private fields to differentiate from local variables
+        // - Constants are also just plain fields without special naming (underscore when private, camel case when public).
+        private int _elapsedTimeInDays;
+        private const int _hoursPerDay = 24;
+
+        // Use [SerializeField] attribute if you want to display a private field in Inspector.
+        // Booleans ask a question that can be answered true or false.
+        [SerializeField] private bool _isPlayerDead;
+
+        // This groups data from the custom PlayerStats class in the Inspector.
+        [SerializeField] private PlayerStats _stats;
+
+        // This limits the values to a Range and creates a slider in the Inspector.
+        [Range(0f, 1f)] [SerializeField] private float _rangedStat;
+
+        // A tooltip can replace a comment on a serialized field and do double duty.
+        [Tooltip("This is another statistic for the player.")]
+        [SerializeField] private float _anotherStat;
```

Comment:
We should mention Unity-style for inspector fields here (camelCase). Do we adhere to it or do we ignore it and go with `PascalCase` anyways?

Reply:
What do you mean? The example above leverages UnityFields the same as non-Unity fields. What is default and what would change in here?
```[SerializeField] private float _anotherStat;```

Reply:
Ah did I misunderstand? Are we going to use private members even for properties set through the inspector?

Reply:
I didn't include this as a recommendation. What's your recommendation? Then I can add this information.

Reply:
Ok. I added another hint at this. Anything else needed? If not, feel free to accept the PR.

```c#
        // Use [SerializeField] attribute if you want to display a private field in Inspector.
        // Try to use private Fields for Components/MonoBehaviours, if you don't need them to be public.
        // Booleans ask a question that can be answered true or false.
        [SerializeField] private bool _isPlayerDead;
```